plugins/module_utils/_api_data.py

```diff
@@ -25,6 +26,7 @@ def __init__(self, primary_keys=None,
             raise ValueError('unknown_mechanism and fully_understood cannot be combined')
         self.primary_keys = primary_keys
         self.stratify_keys = stratify_keys
+        self.require_one_of = require_one_of
```

Comment:
Let's ensure here that we always have a list:
```suggestion
        self.require_one_of = require_one_of or []
```